zerver/data_import/rocketchat.py

```diff
@@ -0,0 +1,673 @@
+        if user_dict["type"] != "user":
+            user_dict["is_mirror_dummy"] = True
+            emails = user_dict.get("emails")
+            if emails is None or len(emails) == 0:
+                user_dict["emails"] = [
+                    {
+                        "address": "{}-{}@{}".format(
+                            user_dict["username"], user_dict["type"], domain_name
+                        )
+                    }
+                ]
+
+        # TODO: Change this to use actual exported avatar
+        avatar_source = "G"
+        full_name = user_dict["name"]
+        id = user_id_mapper.get(rc_user_id)
+        delivery_email = user_dict["emails"][0]["address"]
+        email = user_dict["emails"][0]["address"]
+        short_name = user_dict["username"]
+        date_joined = int(timezone_now().timestamp())
+        timezone = "UTC"
+
+        role = UserProfile.ROLE_MEMBER
+        if "admin" in user_dict["roles"]:
+            role = UserProfile.ROLE_REALM_OWNER
+        elif "guest" in user_dict["roles"]:
+            role = UserProfile.ROLE_GUEST
+
+        if user_dict["is_mirror_dummy"]:
+            is_active = False
+            is_mirror_dummy = True
+        else:
+            is_active = True
+            is_mirror_dummy = False
+
+        user = build_user_profile(
+            avatar_source=avatar_source,
+            date_joined=date_joined,
+            delivery_email=delivery_email,
+            email=email,
+            full_name=full_name,
+            id=id,
+            is_active=is_active,
+            role=role,
+            is_mirror_dummy=is_mirror_dummy,
+            realm_id=realm_id,
+            short_name=short_name,
+            timezone=timezone,
+        )
+        user_handler.add_user(user)
+
+
+def convert_channel_data(
+    room_id_to_room_map: Dict[str, Dict[str, Any]],
+    team_id_to_team_map: Dict[str, Dict[str, Any]],
+    stream_id_mapper: IdMapper,
+    realm_id: int,
+) -> List[ZerverFieldsT]:
+    streams = []
+
+    for rc_room_id in room_id_to_room_map:
+        channel_dict = room_id_to_room_map[rc_room_id]
+
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(rc_room_id)
+        invite_only = channel_dict["t"] == "p"
+
+        stream_name = channel_dict["name"]
+        stream_desc = channel_dict.get("description", "")
+        if channel_dict.get("teamId"):
+            if channel_dict.get("teamMain") is True:
+                stream_name = "[TEAM] " + stream_name
+            else:
+                stream_desc = "[Team {} channel]. {}".format(
+                    team_id_to_team_map[channel_dict["teamId"]]["name"], stream_desc
+                )
+
+        stream = build_stream(
+            date_created=now,
+            realm_id=realm_id,
+            name=stream_name,
+            description=stream_desc,
+            stream_id=stream_id,
+            deactivated=False,
```

Comment:
I think we might need `deactivated=True` for streams with no subscribers, since Zulip auto-deactivates streams in that situation.

Reply:
Done.